File: Navtrack.Core/ClientApp/src/apis/signalR/AssetsHub.ts
import { apiUrl } from "../../services/httpClient/HttpClientUtil";
import { SignalRClient } from "../../services/signalR/SignalRClient";
import { AssetModel } from "../types/asset/AssetModel";

export const AssetsHub = {
  getAll: async (): Promise<AssetModel[]> => {
    let assets = await SignalRClient.invoke<AssetModel[]>(apiUrl("hubs/assets"), "GetAll");

    return assets;
  }
};

File: backend/Navtrack.Api.Shared/Hubs/AssetsHub.cs
using System.Threading.Tasks;
using IdentityServer4;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.SignalR;
using Navtrack.Api.Model.Assets;
using Navtrack.Api.Model.Common;
using Navtrack.Api.Services.Assets;

namespace Navtrack.Api.Shared.Hubs;

[Authorize(IdentityServerConstants.LocalApi.PolicyName)]
public class AssetsHub(IAssetService service) : Hub
{
    public async Task<ListModel<AssetModel>> GetAll()
    {
        ListModel<AssetModel> assets = await service.GetAssets();

        return assets;
    }
}